### DISCUSSION TITLE
determine the name of the .md file used to generate the slides inside the deck

### DISCUSSION THREAD

**Discussion by gureckis** [slidevjs/slidev#1577] [Q&A] [answered]
when I run `slidev` command I might use a file name other than slides.md.  is this exposed anywhere in the api so I could inspect which file it is?  or perhaps it is exposed by Vite?  it seems like at least in dev mode it must because the editor re-writes to the same file but I couldn't find a reference to the file name in `$slidev`.  (sorry for more arcane programming API questions!). use cases are automatically making links to the file in GitHub directly in the slides, or extracting something meta from the filename itself like "lecture01.md" so inside the slides know it is the first lecture.

**Comment by kermanx** [comment]
Hmmm, I don't know how to get this data. How about importing `slidesTitle` from `"@slidev/client/env"` or opening an issue for this feature?

**Comment by gureckis** [comment]
ok thanks!  I assume it is not a major feature that most users need so I won't clog things up with an open issue, but I'll see if I can hack something together and make a PR.

**Comment by gureckis** [comment]
thanks to a comment by @JoeyGrajciar I figured this out.

In `setup/preparser.ts` add the following which injects the file path into the markdown.

```ts
import { definePreparserSetup } from '@slidev/types'

export default definePreparserSetup(({ filepath, headmatter, mode }) => {
  return [
    {
      transformRawLines(lines) {
        // if front matter exists then insert at the top
        // assumes you correctly ended the front matter with '---'
        if (lines[0] === '---') {
          lines.splice(1, 0, `filename: ${filepath}`)
        } else {
          // otherwise add frontmatter to first slide containing the file path
          lines.unshift(`---`)
          lines.unshift(`filename: ${filepath}`)
          lines.unshift(`---`)
        }
      },
    },
  ]
})
```

Then in your slide or vue component you can access via `$slidev.configs.filename`
e.g., `{{ $slidev.configs.filename}}` in a slide.

**Reply by gureckis** [reply, marked as answer]
Here is an updated solution.  The above works but when you open the slide editor in the browser and edit something it writes the entire content of the transformed lines back out to the file.  The previous code was re-inserting the file name multiple times so instead this checks if the filename property exists first before adding it, or otherwise updates it.

The net effect is you could define `filename:` yourself in the front matter but it will be overwritten at launch time with the path to the file

```ts
import { definePreparserSetup } from '@slidev/types'

export default definePreparserSetup(({ filepath, headmatter, mode }) => {
  return [
    {
      transformRawLines(lines) {
        // if front matter exists then insert at the top
        // assumes you correctly ended the front matter with '---'
        if (lines[0] === '---') {
          // check to see if filename is already defined anywhere in the front matter
          let found = false
          for (let i = 1; i < lines.length; i++) {
            if (lines[i].startsWith('filename:')) {
              // if it is, update it
              lines[i] = `filename: ${filepath}`
              found = true
            }
          }
          // otherwise add frontmatter to first slide containing the file path
          if (!found) {
            lines.splice(1, 0, `filename: ${filepath}`)
          }
        } else {
          // otherwise add frontmatter to first slide containing the file path
          lines.unshift(`---`)
          lines.unshift(`filename: ${filepath}`)
          lines.unshift(`---`)
        }
      },
    },
  ]
})

```
